index.js

```diff
@@ -292,27 +304,27 @@ module.exports = class Filter extends Plugin {
     });
   }
 
-  async _handleFile(relativePath, srcDir, destDir, entry, outputPath, forceInvalidation, isChange, stats) {
+  async _handleFile(relativePath, entry, outputPath, forceInvalidation, isChange, stats) {
     stats.handleFile++;
 
     let handleFileStart = process.hrtime();
+    let destDir = this.outputPath + '/' + outputPath;
     try {
       let result;
-      let srcPath = srcDir + '/' + relativePath;
 
       if (this.canProcessFile(relativePath, entry)) {
         stats.processed++;
         if (this._outputLinks[outputPath] === true) {
           delete this._outputLinks[outputPath];
-          fs.unlinkSync(outputPath);
+          fs.unlinkSync(destDir);
```

Comment:
If we don't use `this.output` we also break encapsulation here.

Reply:
Yet to add `unlinkSync` to `this.output`